resources/cluster/resource.go

```diff
@@ -0,0 +1,934 @@
+			HttpStatusCode: configuration.GetConfig()[constants.MongoClientCreationError].Code,
+			HttpError:      configuration.GetConfig()[constants.MongoClientCreationError].Message,
+		}
+	}
+	_, _, projectErr := client.ProjectsApi.GetProject(context.Background(), cast.ToString(inputModel.ProjectId)).Execute()
+	if projectErr != nil {
+		_, _ = logger.Warnf("Get Project error: %v", projectErr.Error())
+		return atlasresponse.AtlasRespone{
+			Response:       nil,
+			HttpStatusCode: configuration.GetConfig()[constants.ProjectDoesNotExist].Code,
+			HttpError:      fmt.Sprintf(configuration.GetConfig()[constants.ProjectDoesNotExist].Message, *inputModel.ProjectId),
+		}
+	}
+
+	/*	endPoints, res, endpointerr := client.PrivateEndpointServicesApi.ListPrivateEndpointServices(ctx, inputModel.ProjectId, "AWS").Execute()
+				if endpointerr != nil {
+			_, _ = logger.Warnf("Get PrivateEndpoint error: %v", endpointerr.Error())
+					return atlasresponse.AtlasRespone{
+						Response:       nil,
+						HttpStatusCode: res.StatusCode,
+						HttpError:      endpointerr.Error(),
+					}
+				}
+
+				count := len(endPoints)
+				if count == 0 {
+		_, _ = logger.Warnf("Get PrivateEndpoint Not Configured error: %v", endpointerr.Error())
+					return atlasresponse.AtlasRespone{
+						Response:       nil,
+						HttpStatusCode: http.StatusInternalServerError,
+						HttpError:      "No Entpoints configured for this project",
+					}
+				}
+				_, _ = logger.Debugf("Cluster create projectId: %s, clusterName: %s", inputModel.ProjectId, inputModel.ClusterName)
+	*/
+	currentModel, err := loadCurrentModel(*inputModel)
+	if err != nil {
+		_, _ = logger.Warnf("Create Current Model error: %v", err.Error())
+		return atlasresponse.AtlasRespone{
+			Response:       nil,
+			HttpStatusCode: configuration.GetConfig()[constants.ClusterModelError].Code,
+			HttpError:      configuration.GetConfig()[constants.ClusterModelError].Message,
+		}
+	}
+	currentModel.validateDefaultLabel()
+
+	// Prepare cluster request
+	clusterRequest, err := setClusterRequest(&currentModel)
+	if err != nil {
+		_, _ = logger.Warnf("Create Cluster Request error: %v", err.Error())
+		return atlasresponse.AtlasRespone{
+			Response:       nil,
+			HttpStatusCode: configuration.GetConfig()[constants.ClusterRequestError].Code,
+			HttpError:      configuration.GetConfig()[constants.ClusterRequestError].Message,
+		}
+	}
+
+	// Create Cluster
+	cluster, _, err := client.MultiCloudClustersApi.CreateCluster(ctx, cast.ToString(currentModel.ProjectId), clusterRequest).Execute()
+
+	if err != nil {
+		_, _ = logger.Warnf("Create - Cluster.Create() - error: %+v", err)
+		return atlasresponse.AtlasRespone{
+			Response:       nil,
+			HttpStatusCode: configuration.GetConfig()[constants.ClusterCreateError].Code,
+			HttpError:      configuration.GetConfig()[constants.ClusterCreateError].Message,
+		}
+	}
+	currentModel.StateName = cluster.StateName
+	return atlasresponse.AtlasRespone{
+		Response:       nil,
```

Comment:
why the response is nil.we can return configuration details along with cluster name